lock

```diff
@@ -1,70 +1,90 @@
+OUTPUT_IMG_HEIGHT=0  # Decide size to cover all screens
+
+# i3lock command
+LOCK_BASE_CMD="i3lock -i $OUTPUT_IMG"
+if [ "$lock_args" ]; then
+	LOCK_ARGS="$lock_args"  # Passed command
+else
+	LOCK_ARGS="-t -p default -e"  # Default
+fi
+LOCK_CMD="$LOCK_BASE_CMD $LOCK_ARGS"
 
-if [ -e $OUTPUT_IMG ]
-then
-    # Lock screen since image already exists
-    i3lock --image=$OUTPUT_IMG --tiling --show-failed-attempts
-    exit 0
+# If the image already exists (in cache) then lock the screen and exit:
+if [ -e $OUTPUT_IMG ]; then
+	$LOCK_CMD
+	exit 0
 fi
 
-#Execute xrandr to get information about the monitors:
-while read LINE
-do
-  #If we are reading the line that contains the position information:
-  if [[ $LINE =~ $DISPLAY_RE ]]; then
-    #Extract information and append some parameters to the ones that will be given to ImageMagick:
-    SCREEN_WIDTH=${BASH_REMATCH[1]}
-    SCREEN_HEIGHT=${BASH_REMATCH[2]}
-    SCREEN_X=${BASH_REMATCH[3]}
-    SCREEN_Y=${BASH_REMATCH[4]}
+# Execute xrandr to get information about the monitors:
+#while read LINE
+#do
+while IFS= read -r LINE; do
+	# If we are reading the line that contains the position information:
+	if [[ $LINE =~ $DISPLAY_RE ]]; then
+		# Extract information and append some parameters to the ones that will be given to ImageMagick:
+		SCREEN_WIDTH=${BASH_REMATCH[1]}
+		SCREEN_HEIGHT=${BASH_REMATCH[2]}
+		SCREEN_X=${BASH_REMATCH[3]}
+		SCREEN_Y=${BASH_REMATCH[4]}
     
-    CACHE_IMG="$CACHE_FOLDER""$SCREEN_WIDTH"x"$SCREEN_HEIGHT"."$MD5_BKG_IMG"."$MD5_TEXT_IMG".png
-    ## if cache for that screensize doesnt exist
-    if ! [ -e $CACHE_IMG ]
-    then
-	# if this is the first time cache was not hit then
-	if ! [ $(echo $TEXT_IMG_INFO) ]
-	then	
-	    #Get dimensions of the text image
-	    TEXT_IMG_INFO=`identify $TEXT_IMG`
-	    [[ $TEXT_IMG_INFO =~ $IMAGE_RE ]]
-	    TEXT_IMG_WIDTH=${BASH_REMATCH[1]}
-	    TEXT_IMG_HEIGHT=${BASH_REMATCH[2]}
-        fi
-	# Create image for that screensize
-        TEXT_X=$(($X+$SCREEN_WIDTH/2-$TEXT_IMG_WIDTH/2))
-        TEXT_Y=$(($Y+$SCREEN_HEIGHT/2-$TEXT_IMG_HEIGHT/2+200))
-        eval convert '\(' '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}!' '\)' \
-		'\(' '$TEXT_IMG' '-geometry' '+$TEXT_X+$TEXT_Y' '\)' '-composite' '$CACHE_IMG'
-    fi
+		CACHE_IMG="$CACHE_FOLDER""$SCREEN_WIDTH"x"$SCREEN_HEIGHT"."$MD5_BKG_IMG".png
+		# If cache for that screensize doesnt exist
+		if ! [ -e $CACHE_IMG ]; then
+			#eval convert '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}!' '$CACHE_IMG'
+			eval convert '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}^' '-gravity' 'Center' '-crop' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}+0+0' '+repage' '$CACHE_IMG'
+		fi
 
-    # Decide size of output image
-    if (( $OUTPUT_IMG_WIDTH < $SCREEN_WIDTH+$SCREEN_X )); then OUTPUT_IMG_WIDTH=$(($SCREEN_WIDTH+$SCREEN_X)); fi;
-    if (( $OUTPUT_IMG_HEIGHT < $SCREEN_HEIGHT+$SCREEN_Y )); then OUTPUT_IMG_HEIGHT=$(( $SCREEN_HEIGHT+$SCREEN_Y )); fi;
+		# Decide size of output image
```

Comment:
whitespace added

Reply:
What do you mean? I just don't like to use white spaces instead of tabs (which I'll change back to minimize the diff). Where do I add a white space?

Reply:
Its okay to use tabs if you like but when working with existing code you have to keep it consistent and go with whatever is being used. I just wanted to point out that this line is just whitespace change not anything functional.